prow/cmd/deck/main.go

```diff
@@ -147,6 +149,30 @@ func loadToken(file string) (string, error) {
 	return string(bytes.TrimSpace(raw)), nil
 }
 
+// copy a http.Request
+// see: https://go-review.googlesource.com/c/go/+/36483/3/src/net/http/server.go
+func dupeRequest(original *http.Request) *http.Request {
+	r2 := new(http.Request)
+	*r2 = *original
+	r2.URL = new(url.URL)
+	*r2.URL = *original.URL
+	return r2
+}
+
+// serve with handler but map extensionless URLs to to the default
+func defaultExtension(extension string, h http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if len(r.URL.Path) > 0 &&
+			r.URL.Path[len(r.URL.Path)-1] != '/' && path.Ext(r.URL.Path) == "" {
```

Comment:
```go
path.Ext("somepath/") == ""
```

So the trailing slash check is superfluous

Reply:
This differentiates between `foo/` and `foo/bar`, so you can still serve directories with an index. we don't want to serve `foo/.html`...

Reply:
The idea is to only add a default extension to paths that look like serving a specific file like `/foo`, `/bar/baz` and not `/bar/` or `/`.

Reply:
Ah, right. ACK